src/blocks/FilterBlock/FilterBlock.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {useMemo, useState} from 'react';
+
+import {block} from '../../utils';
+import {BlockType, ConstructorItem, FilterBlockProps, FilterItem} from '../../models';
+import {Row, Col} from '../../grid';
+import {BlockHeader, AnimateBlock} from '../../components';
+import ButtonTabs, {ButtonTabsItemProps} from '../../components/ButtonTabs/ButtonTabs';
+import {ConstructorBlocks} from '../../containers/PageConstructor/components/ConstructorBlocks';
+
+import './FilterBlock.scss';
+
+const b = block('filter-block');
+const DEFAULT_ALL_TAG_TITLE = 'All';
+
+const FilterBlock: React.FC<FilterBlockProps> = ({
+    title,
+    description,
+    filterTags,
+    tagSize,
+    allTag,
+    items,
+    colSizes,
+    centered,
+    animated,
+}) => {
+    const tabButtons = useMemo(() => {
+        const allButton: ButtonTabsItemProps | undefined = allTag
+            ? {id: null, title: typeof allTag === 'boolean' ? DEFAULT_ALL_TAG_TITLE : allTag}
+            : undefined;
+        const otherButtons: ButtonTabsItemProps[] | undefined =
+            filterTags && filterTags.map((tag) => ({id: tag.id, title: tag.label}));
+        return [...(allButton ? [allButton] : []), ...(otherButtons ? otherButtons : [])];
```

Comment:
Unfortunately cannot change this line as suggested, because we merge here two arrays to omit nulls in final array. Replacing ternary operator with the ?? operator because in this case type of result wont be an array.